tailscale/client.go

```diff
@@ -359,25 +360,49 @@ func (c *Client) DeviceSubnetRoutes(ctx context.Context, deviceID string) (*Devi
 	return &resp, nil
 }
 
+// MaybeEmptyTime wraps a time and allows for unmarshalling timestamps that represent an empty time as an empty string (e.g "")
+// this is used by the tailscale API when it returns services that have no created date, such as it's hello service.
+type MaybeEmptyTime struct {
+	time.Time
+}
+
+// MarshalJSON is an implementation of json.Marshal.
+func (t MaybeEmptyTime) MarshalJSON() ([]byte, error) {
+	return json.Marshal(t.Time)
+}
+
+// MarshalJSON returns nil if it is a blank string, otherwise, returns the result of json.Unmarshal.
+func (t *MaybeEmptyTime) UnmarshalJSON(data []byte) error {
+	if string(data) == "\"\"" {
```

Comment:
```suggestion
	if string(data) == `""` {
```

Just a bit easier to read and you don't need to escape things